### PR TITLE
list users

### PR DESCRIPTION
**Pull request by candelarialopezz**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -26,6 +26,10 @@ class User < ApplicationRecord
 
   has_many :posts
 
+  paginates_per 5
+  max_paginates_per 15
+  max_pages 10
```
Comment (daizuoheng) [inline app/models/user.rb:31]: Let’s move all configurations into the kaminari initializer.

File: app/views/users/_side_user_data.html.haml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+- if user
```
Comment (daizuoheng) [inline app/views/users/_side_user_data.html.haml:1]: This line is unnecessary, you already checked it when you call this partial.

File: config/routes.rb [hunk 3]
```diff
@@ -1,14 +1,9 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
+  get '/users', to: 'users#index', as: 'users'
```
Comment (daizuoheng) [inline config/routes.rb:4]:
```suggestion
  resources :users, only: %i[index]
```

File: app/views/users/show.html.haml [hunk 4]
```diff
@@ -1,25 +1,10 @@
...
-      = current_user.last_name
-      %br
-      = current_user.birthday.strftime('%d/%m/%Y')
-      %br
-      %br
-      = 'Posts: '
-      = @posts.count
-      %br
-      = link_to 'Edit', edit_user_registration_path, class: 'btn btn-primary btn-newPost'
+- if @user && @user.nickname.present?
```
Comment (daizuoheng) [inline app/views/users/show.html.haml:1]: This line is also unnecessary. If user doesn’t exist, the controller will redirect into root page.

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -2,10 +2,15 @@
...
-      @posts = current_user.posts.order(published_at: :desc)
-    else
-      redirect_to root_path
-    end
+    @user = User.find_by(nickname: params[:nickname])
+    @posts = @user.posts.order(published_at: :desc) if @user
+
+    return if @user
+
+    redirect_to root_path, alert: 'User not found'
```
Comment (EnzoCozza) [inline app/controllers/users_controller.rb:10]: Let's re-order this so it's clearer
```suggestion
    redirect_to root_path, alert: 'User not found' unless @user

    @posts = @user.posts.order(published_at: :desc)
```

File: spec/requests/users_spec.rb
Comment (daizuoheng) [inline spec/requests/users_spec.rb:1]: You must create more than 1 user to ensure the indexing is functioning properly.
You should first test that the initial page displays the correct number per page.
Next, confirm that the first page shows the nicknames of the first X users.
Ensure that the last page displays the remaining users.
Finally, verify that the last page contains the nicknames of the last X users.

File: app/controllers/posts_controller.rb [hunk 6]
```diff
@@ -19,6 +19,7 @@ def show
   end
 
   def index
+    @user = User.find_by(nickname: params[:nickname])
```
Comment (daizuoheng) [inline app/controllers/posts_controller.rb:22]: Why are you adding this line?

File: app/views/users/_user.html.haml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+- user = User.find_by(nickname: user.nickname)
```
Comment (EnzoCozza) [inline app/views/users/_user.html.haml:1]: `user` should already be loaded as it's a local variable

File: app/views/users/_user.html.haml [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+
+- if user
```
Comment (EnzoCozza) [inline app/views/users/_user.html.haml:3]: You shouldn't need this validation as you checked this before

File: app/controllers/users_controller.rb [hunk 9]
```diff
@@ -2,10 +2,14 @@
...
 class UsersController < ApplicationController
   def show
-    if params[:nickname] == current_user.nickname
-      @posts = current_user.posts.order(published_at: :desc)
-    else
-      redirect_to root_path
-    end
+    @user = User.find_by(nickname: params[:nickname])
+
+    redirect_to root_path, alert: 'User not found' unless @user
```
Comment (EnzoCozza) [inline app/controllers/users_controller.rb:7]: You need to add a return here, else you will have redirection issues

File: app/controllers/users_controller.rb [hunk 10]
```diff
@@ -2,10 +2,14 @@
...
+
+    @posts = @user.posts.order(published_at: :desc) if @user
```
Comment (EnzoCozza) [inline app/controllers/users_controller.rb:9]: And you should remove the `if` here as you are checking this in the previous line

File: spec/requests/users_spec.rb [hunk 11]
```diff
@@ -138,4 +138,92 @@
...
+  describe '#index_user_request' do
+    image_path = File.join(File.dirname(__FILE__), '..', 'images', 'download.jpeg')
+    let(:user1) { create :user, password: 'password' }
+    let(:user2) { create :user, password: 'password' }
+    let(:user3) { create :user, password: 'password' }
+    let(:user4) { create :user, password: 'password' }
+    let(:user5) { create :user, password: 'password' }
+    let(:user6) { create :user, password: 'password' }
+    let(:user7) { create :user, password: 'password' }
+    let(:user8) { create :user, password: 'password' }
```
Comment (EnzoCozza) [inline spec/requests/users_spec.rb:151]: Let's use create_list for this!

File: spec/requests/users_spec.rb [hunk 12]
```diff
@@ -138,4 +138,92 @@
...
+
+    before do
+      user1.avatar.attach(io: File.open(image_path), filename: 'download.jpeg', content_type: 'image/jpeg')
+      user2.avatar.attach(io: File.open(image_path), filename: 'download.jpeg', content_type: 'image/jpeg')
+      user3.avatar.attach(io: File.open(image_path), filename: 'download.jpeg', content_type: 'image/jpeg')
+      user4.avatar.attach(io: File.open(image_path), filename: 'download.jpeg', content_type: 'image/jpeg')
+      user5.avatar.attach(io: File.open(image_path), filename: 'download.jpeg', content_type: 'image/jpeg')
+      user6.avatar.attach(io: File.open(image_path), filename: 'download.jpeg', content_type: 'image/jpeg')
+      user7.avatar.attach(io: File.open(image_path), filename: 'download.jpeg', content_type: 'image/jpeg')
+      user8.avatar.attach(io: File.open(image_path), filename: 'download.jpeg', content_type: 'image/jpeg')
```
Comment (EnzoCozza) [inline spec/requests/users_spec.rb:164]: And this should be done through an each

**Review by EnzoCozza — APPROVED** [review]

**Review by daizuoheng — APPROVED** [review]

**Pull request merged**